universal-profiling-integration/src/test/java/co/elastic/otel/UniversalProfilingProcessorTest.java

```diff
@@ -514,6 +514,7 @@ void shutdownFlushesBufferedSpans() {
 
       try (OpenTelemetrySdk sdk = initSdk(builder -> builder.clock(() -> 0L))) {
         sendProfilerRegistrationMsg(1, "hostid");
+        processor.pollMessagesAndFlushPendingSpans();
```

Comment:
This change fixes an actually unrelated flakyness: The test assumes that the profielr registration message has been consumed, but this isn't guaranteed by the background polling. For this reason we have to force the span processor to explicitly poll and process the messages.